src/datasource/niftizarr/frontend.ts

```diff
@@ -0,0 +1,205 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file Support for displaying single NIfTI Zarr (https://github.com/lincbrain/linc-docs/blob/dd3411c15643df2662a26dee752a394dfc282872/dev/nifti_zarr_format.md)
+ * files as volumes.
+ */
+
+import {
+  getMetadata,
+  ZarrMultiscaleInfo,
+  resolveOmeMultiscale,
+  getMultiscaleInfoForSingleArray,
+  MultiscaleVolumeChunkSource
+} from "#src/datasource/zarr/frontend.ts";
+import {
+  getNiftiVolumeInfo,
+} from "#src/datasource/nifti/frontend.ts";
+import {
+  makeCoordinateSpace,
+  makeIdentityTransformedBoundingBox,
+} from "#src/coordinate_transform.js";
+
+import "#src/datasource/zarr/codec/blosc/resolve.js";
+import "#src/datasource/zarr/codec/zstd/resolve.js";
+
+import { makeDataBoundsBoundingBoxAnnotationSet } from "#src/annotation/index.js";
+import type {
+  CompleteUrlOptions,
+  DataSource,
+  GetDataSourceOptions,
+} from "#src/datasource/index.js";
+import { DataSourceProvider } from "#src/datasource/index.js";
+import "#src/datasource/zarr/codec/bytes/resolve.js";
+import "#src/datasource/zarr/codec/crc32c/resolve.js";
+import "#src/datasource/zarr/codec/gzip/resolve.js";
+import "#src/datasource/zarr/codec/sharding_indexed/resolve.js";
+import "#src/datasource/zarr/codec/transpose/resolve.js";
+import {
+  parseDimensionSeparator,
+} from "#src/datasource/zarr/metadata/parse.js";
+import { parseOmeMetadata } from "#src/datasource/zarr/ome.js";
+import { completeHttpPath } from "#src/util/http_path_completion.js";
+import {
+  parseQueryStringParameters,
+  verifyObject,
+  verifyOptionalObjectProperty,
+} from "#src/util/json.js";
+import {
+  parseSpecialUrl,
+} from "#src/util/special_protocol_request.js";
+import { uncancelableToken } from "#src/util/cancellation.js";
+
+
+export class NiftiZarrDataSource extends DataSourceProvider {
+  constructor(public zarrVersion: 2 | 3 | undefined = undefined) {
+    super();
+  }
+  get description() {
+    const versionStr =
+      this.zarrVersion === undefined ? "" : ` v${this.zarrVersion}`;
+    return `Nifti Zarr${versionStr} data source`;
+  }
```

Comment:
Do you want to default to `undefined`, rather than something like `1`, etc.?

`registerProvider("niftizarr", () => new NiftiZarrDataSource());` -- will return `undefined` as a version just as an FYI

Reply:
i copied from zarr frontend.... that's how they defaulted i left it the same just added the word nifti